src/tests/daosctl/test-pool.c

```diff
@@ -167,6 +167,9 @@ cmd_connect_pool(int argc, const char **argv, void *ctx)
 	 */
 	argp_parse(&argp, argc, (char **restrict)argv, 0, 0, &cp_options);
 
+	if (cp_options.uuid == NULL)
```

Comment:
Line 181 calls uuid_parse.  Should we also not check the return value there?  The uuid could be NULL but also could be garbage

Reply:
Good point, should return if rc < 0.
Just clarify, we still need to check uuid NULL before uuid_parse, since uuid_parse doesn't handle NULL argument(strlen(NULL) will segment fault).

Reply:
I’d probably return EINVAL there too

Reply:
hmm, agree after checking back the caller process_cmd() that only print out the help msg on EINVAL rc:
        if (rc == EINVAL) {
                D_PRINT("Unknown command or missing argument: %s\n\n", argv[1]);
                print_help();
        }

Reply:
Also put the same check in cont-cmds.c for cmd_create_container().